From 97208d22d1f9bc1fc9fa6b8b64fbbbeacdedb73c Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Tue, 8 Sep 2020 10:01:45 -0700
Subject: [PATCH 4/4] Pass --disable-reloc-section on PE targets for PR 25662
 test

Pass --disable-reloc-section on PE targets for PR 25662 test since

commit 514b4e191d5f46de8e142fe216e677a35fa9c4bb

Author: Jeremy Drake <sourceware-bugzilla@jdrake.com>
Date:   Thu Aug 27 12:58:27 2020 +0100

    Change the default characteristics of DLLs built by the linker to more secure settings.

defaulted to --enable-reloc-section.

	PR ld/26587
	* testsuite/binutils-all/objcopy.exp: Pass --disable-reloc-section
	to ld on PE targets for PR 25662 test.
---
 binutils/ChangeLog                          |  6 ++++++
 binutils/testsuite/binutils-all/objcopy.exp | 10 +++++++++-
 2 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/binutils/ChangeLog b/binutils/ChangeLog
index 2b9ac6e792..678a19f572 100644
--- a/binutils/ChangeLog
+++ b/binutils/ChangeLog
@@ -1,3 +1,9 @@
+2020-09-08  H.J. Lu  <hongjiu.lu@intel.com>
+
+	PR ld/26587
+	* testsuite/binutils-all/objcopy.exp: Pass --disable-reloc-section
+	to ld on PE targets for PR 25662 test.
+
 2020-07-24  Nick Clifton  <nickc@redhat.com>
 
 	2.35 Release:
diff --git a/binutils/testsuite/binutils-all/objcopy.exp b/binutils/testsuite/binutils-all/objcopy.exp
index dd74860f9e..90ababf905 100644
--- a/binutils/testsuite/binutils-all/objcopy.exp
+++ b/binutils/testsuite/binutils-all/objcopy.exp
@@ -1354,4 +1354,12 @@ if { [istarget pdp11-*-*] } {
     set src "pr25662.s"
 }
 
-objcopy_test "pr25662" $src executable "" "-T$srcdir/$subdir/pr25662.ld"
+set ldflags "-T$srcdir/$subdir/pr25662.ld"
+if { [istarget *-*-cygwin] || [istarget *-*-mingw*] } {
+   append ldflags " --disable-reloc-section"
+}
+
+#xcoff doesn't support arbitrary sections
+if { ![is_xcoff_format] } {
+    objcopy_test "pr25662" $src executable "" $ldflags
+}
-- 
2.28.0.windows.1

